File: bac_manager/bac_manager_interface/src/main/java/com/bac/manager/service/BrandService.java
package com.bac.manager.service;

import com.bac.pojo.TbBrand;
import com.bac.utils.PageResult;

import java.util.List;
import java.util.Map;

public interface BrandService {


    /*
    需求:查询所有品牌数据
     */
    public List<TbBrand> findAll();

    /*
    返回分页列表
     */
    public PageResult findPage(int pageNum, int pageSize);

    /*
    需求:插入品牌数据
     */
    public void add(TbBrand brand);

    /*
    需求:根据id查品牌,用于回显
     */
    public TbBrand findOne(Long id);

    /*
    需求:修改品牌信息.
     */
    public void update(TbBrand brand);

    /*
    需求:根据id删除品牌数据
     */
    public void delete(long[] ids);

    /*
    需求:查询品牌下拉列表
    查询数据格式:[{id:'1',text:'联想'},{id:'2',text:'华为'}]
    返回值:List<Map>
     */
    public List<Map> findBrandList();


}

File: bac_shop_web/src/main/webapp/js/service/brandService.js
//定义service服务层
app.service("brandService", function ($http) {
    //把内置服务发送请求方法全部抽取到服务层代码

    //查询所有
    this.findAll = function () {
        return $http.get("../brand/findAll");
    };
    //分页查询
    this.findPage = function (pageNum, pageSize) {
        return $http.get("../brand/findPage/" + pageNum + "/" + pageSize);
    };
    //添加
    this.add = function (entity) {
        return $http.post("../brand/add", entity);
    };
    //修改
    this.update = function (entity) {
        return $http.post("../brand/update", entity);
    };
    //批量删除
    this.dele = function (ids) {
        return $http.get("../brand/delete/" + ids);
    };
    //根据id查询
    this.findOne = function (id) {
        return $http.get("../brand/findOne?id=" + id);
    };

    /*
    需求:查询品牌下拉列表
    查询数据格式:[{id:'1',text:'联想'},{id:'2',text:'华为'}]
    返回值:List<Map>
     */
    this.findBrandList = function () {
        return $http.get("../brand/findBrandList");
    }


});